storage_service/locations/models/dataverse.py

```diff
@@ -112,63 +212,113 @@ def move_to_storage_service(self, src_path, dest_path, dest_space):
+                dataset, f, sort_keys=True, indent=4, separators=(',', ': '))
 
         # Fetch all files in dataset.json
-        for file_entry in dataset['latestVersion']['files']:
-            entry_id = str(file_entry['dataFile']['id'])
-            if not file_entry['label'].endswith('.tab'):
-                download_path = os.path.join(dest_path, file_entry['dataFile']['filename'])
-                url = 'https://' + self.host + '/api/access/datafile/' + entry_id
+        for file_entry in dataset["latestVersion"]["files"]:
+            zipped_bundle = False
+            entry_id = str(file_entry["dataFile"]["id"])
+            if not file_entry["label"].endswith(".tab"):
+                download_path = os.path.join(
+                    dest_path, file_entry["dataFile"]["filename"]
+                )
+                url = "https://{}/api/access/datafile/{}".format(
+                    self.host, entry_id
+                )
             else:
-                # If the file is the tab file, download the bundle instead
-                download_path = os.path.join(dest_path, file_entry['label'][:-4] + '.zip')
-                url = 'https://' + self.host + '/api/access/datafile/bundle/' + entry_id
-            LOGGER.debug('URL: %s, params: %s', url, params)
-            response = requests.get(url, params=params)
-            LOGGER.debug('Response: %s', response)
-            with open(download_path, 'wb') as f:
-                f.write(response.content)
+                # If the file is a tab file, download the bundle instead.
+                #
+                # A table based dataset ingested into Dataverse is called a
+                # Tabular Data File. The .tab file format is downloaded as a
+                # 'bundle' from Dataverse. This bundle provides multiple
+                # representations in different formats of the same data.
+                #
+                # A bundle has the property of being a zip file when pulled
+                # down by the storage service. We want to extract this bundle
+                # below, and allow Archivematica to process other zip files as
+                # it would normally (as configured) in the transfer workflow.
+                #
+                # Integrity checks are completed by the Dataverse
+                # microservices.
+                zipped_bundle = True
+                download_path = os.path.join(
+                    dest_path, file_entry["label"][:-4] + ".zip"
+                )
+                url = "https://{}/api/access/datafile/bundle/{}".format(
+                    self.host, entry_id
+                )
+            LOGGER.debug("URL: %s, params: %s", url, params)
+            response = requests.get(url, params=params, stream=True)
+            with open(download_path, "wb") as f:
+                for chunk in response.iter_content(8192):
+                    f.write(chunk)
+            if zipped_bundle and ".zip" in download_path:
```

Comment:
Isn't the second clause in the condition redundant here? I mean, you just added the ".zip" extension in the only path where `zipped_bundle` gets set to `True`, so it's not possible for one condition to be true but not the other.

Reply:
:+1: